### PR TITLE
Custom checks with Pinia store

### PR DESCRIPTION
**Pull request by mikeminutillo**
Moves the custom checks data management to a pinia store.

- [x] Custom checks page
- [x] Menu Item
- [x] Dashboard badge
- [ ] ~~Updates ServiceControl connection status~~ Other components are still doing this on a regular cadence so it is fine if one of them stops (for now).
- [x] Handles deletes

### CODE REVIEW

**Review by PhilBastian — APPROVED** [review]

File: src/ServicePulse.Host/vue/src/stores/CustomChecksStore.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import CustomCheck from "@/resources/CustomCheck";
+import { acceptHMRUpdate, defineStore } from "pinia";
+import { ref, watch } from "vue";
+import useAutoRefresh from "@/composables/autoRefresh";
+
+export const useCustomChecksStore = defineStore("CustomChecksStore", () => {
+  const prefix = "customchecks/";
+
+  const pageNumber = ref(1);
+  const failingCount = ref(0);
```
Comment (PhilBastian) [inline src/ServicePulse.Host/vue/src/stores/CustomChecksStore.ts:11]: could be a calculated value of failedChecks.length

File: src/ServicePulse.Host/vue/src/stores/CustomChecksStore.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      failedChecks.value = [];
+      failingCount.value = 0;
+      throw e;
+    }
+  }, 5000);
+
+  watch(pageNumber, () => dataRetriever.executeAndResetTimer());
+
+  async function dismissCustomCheck(id: string) {
+    dataRetriever.executeAndResetTimer(async () => {
```
Comment (PhilBastian) [inline src/ServicePulse.Host/vue/src/stores/CustomChecksStore.ts:29]: probably want a comment here about the possibility of a zombie check appearing after next refresh if delete hasn't completed yet

File: src/ServicePulse.Host/vue/src/stores/CustomChecksStore.ts (hunk 1 above)
Comment (mikeminutillo) [reply src/ServicePulse.Host/vue/src/stores/CustomChecksStore.ts:11]: It isn't that though. `failedChecks` is a page of data. `failingCount` is the total, irrespective of paging. It is used to figure out the number of pages.

File: src/ServicePulse.Host/vue/src/composables/autoRefresh.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  }
+
+  function startTimer() {
+    stopTimer();
+    refreshInterval = window.setTimeout(() => {
+      executeAndResetTimer();
+    }, timeout);
+  }
+
+  async function executeAndResetTimer(overrideAction: (() => Promise<void>) | null = null) {
```
[the remark this answers is not in the document]
Comment (mikeminutillo) [reply src/ServicePulse.Host/vue/src/composables/autoRefresh.ts:18]: I had this but it changes the type to `(() => Promise<void>) | undefined`. Is that an issue?

File: src/ServicePulse.Host/vue/src/components/customchecks/CustomChecksMenuItem.vue [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+<template>
+  <RouterLink :class="{ active: isRouteSelected(routeLinks.customChecks) }" :to="routeLinks.customChecks">
+    <i class="fa fa-check icon-white" title="Custom Checks"></i>
+    <span class="navbar-label">Custom Checks</span>
+    <span v-if="failingCount > 0" class="badge badge-important">{{ failingCount }}</span>
+  </RouterLink>
+</template>
+
+<style scoped>
+@import "@/assets/navbar.css";
```
[the remark this answers is not in the document]
Comment (mikeminutillo) [reply src/ServicePulse.Host/vue/src/components/customchecks/CustomChecksMenuItem.vue:21]: Is there a reason they aren't in `navbar.css`? Is that used in other contexts?

File: src/ServicePulse.Host/vue/src/components/customchecks/CustomCheckView.vue [hunk 5]
```diff
@@ -40,15 +31,15 @@ async function dismissCustomCheck(id: string) {
...
+          <button type="button" class="btn btn-default" title="Dismiss this custom check so it doesn't show up as an alert" @click="store.dismissCustomCheck(customCheck.id)">Dismiss</button>
         </div>
       </div>
     </div>
   </div>
 </template>
 
 <style scoped>
-@import "./list.css";
+@import "../list.css";
```
Comment (TravisNickels) [inline src/ServicePulse.Host/vue/src/components/customchecks/CustomCheckView.vue:42]:
```suggestion
@import "@/components/list.css";
```
Just for consistency?

**Review by TravisNickels — APPROVED** [review]

File: src/ServicePulse.Host/vue/src/composables/autoRefresh.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (PhilBastian) [reply src/ServicePulse.Host/vue/src/composables/autoRefresh.ts:18]: not if you treat null and undefined the same, which we almost always do

File: src/ServicePulse.Host/vue/src/components/customchecks/CustomChecksMenuItem.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (PhilBastian) [reply src/ServicePulse.Host/vue/src/components/customchecks/CustomChecksMenuItem.vue:21]: navbar.css i think was originally from a bootstrap context and therefore potentially has a lot of useless styles which could eventually be extracted, whereas these styles were specific to what we were trying to achieve

[the remark this answers is not in the document]
Comment (PhilBastian) [reply src/ServicePulse.Host/vue/src/components/customchecks/CustomChecksMenuItem.vue:21]: and yes navbar.css is used in at least one other context

**Pull request merged**